synapse/models/infotech.py

```diff
@@ -1,7 +1,150 @@
+import synapse.common as s_common
+import synapse.datamodel as s_datamodel
+
+import synapse.lib.version as s_version
+
 from synapse.lib.module import CoreModule, modelrev
+from synapse.lib.types import DataType
+
+
+class SemverType(DataType):
+    subprops = (
+        ('major', {'ptype': 'int'},),
+        ('minor', {'ptype': 'int'},),
+        ('patch', {'ptype': 'int'},),
+        ('build', {'ptype': 'str:txt'},),
+        ('prerelease', {'ptype': 'str:txt'},),
+    )
+
+    def norm(self, valu, oldval=None):
+        if isinstance(valu, int):
+            return self._norm_int(valu, oldval=oldval)
+
+        if isinstance(valu, str):
+            return self._norm_str(valu, oldval=oldval)
+
+        self._raiseBadValu(valu, mesg='Invalid type encountered when norming a semver', type=type(valu))
+
+    def _norm_str(self, text, oldval=None):
+        text = text.strip()
+        if not text:
+            self._raiseBadValu(text, mesg='No text left after stripping whitespace')
+
+        subs = s_version.parseSemver(text)
+        if subs is None:
+            self._raiseBadValu(text, mesg='Unable to parse string as a semver.')
+        valu = s_version.packVersion(subs.get('major'), subs.get('minor'), subs.get('patch'))
+        return valu, subs
+
+    def _norm_int(self, valu, oldval=None):
+        if valu < 0:
+            self._raiseBadValu(valu, mesg='Cannot norm a negative integer as a semver.')
+        if valu > s_version.mask96:
+            self._raiseBadValu(valu, mesg='Cannot norm a integer larger than our version string mask as a semver.')
```

Comment:
I might rename this from `our version string mask` to whatever it actually is, explicitly.